AdobeStockAsset/Model/AssetRepository.php

```diff
@@ -63,21 +68,39 @@ class AssetRepository implements AssetRepositoryInterface
     private $searchResultFactory;
 
     /**
-     * @var AssetKeywordRepositoryInterface
+     * @var GetAssetKeywordsInterface
      */
-    private $assetKeywordRepository;
+    private $getAssetKeywords;
+
+    /**
+     * @var SaveAssetKeywordsInterface
+     */
+    private $saveAssetKeywords;
+
+    /**
+     * @var SaveMediaAssetInterface
+     */
+    private $saveMediaAsset;
+
+    /**
+     * @var MediaAssetInterfaceFactory
+     */
+    private $mediaAssetInterfaceFactory;
 
     /**
      * AssetRepository constructor.
      *
-     * @param ResourceModel $resource
-     * @param AssetCollectionFactory $collectionFactory
-     * @param AssetFactory $factory
-     * @param JoinProcessorInterface $joinProcessor
-     * @param CollectionProcessorInterface $collectionProcessor
+     * @param ResourceModel                      $resource
```

Comment:
Please, avoid using tab formatting as developers always will have to change it during constructor refactoring.